### ISSUE TITLE
[CI] CacheFileTests.testCacheFileCreatedAsSparseFile fails on master

### ISSUE THREAD

**Issue by przemekwitek** [elastic/elasticsearch#81362]
**Build scan**:
https://gradle-enterprise.elastic.co/s/lny65mo6chmek

**Repro line**:
```
REPRODUCE WITH: ./gradlew ':x-pack:plugin:searchable-snapshots:test' --tests "org.elasticsearch.xpack.searchablesnapshots.cache.common.CacheFileTests.testCacheFileCreatedAsSparseFile" \
  -Dtests.seed=E7AB31A9E717E54B \
  -Dtests.locale=zh-CN \
  -Dtests.timezone=Etc/GMT-5 \
  -Druntime.java=17

REPRODUCE WITH: ./gradlew ':x-pack:plugin:searchable-snapshots:test' --tests "org.elasticsearch.xpack.searchablesnapshots.cache.common.CacheFileTests.testCacheFileCreatedAsSparseFile" \
  -Dtests.seed=E7AB31A9E717E54B \
  -Dtests.locale=zh-CN \
  -Dtests.timezone=Etc/GMT-5 \
  -Druntime.java=17
```

**Reproduces locally?**:
No

**Applicable branches**:
`master`

**Failure history**:
Two failures today.
No prior failures of this test.

**Failure excerpt**:
```
11:54:36 org.elasticsearch.xpack.searchablesnapshots.cache.common.CacheFileTests > testCacheFileCreatedAsSparseFile FAILED
11:54:36     java.lang.AssertionError
11:54:36         at __randomizedtesting.SeedInfo.seed([45B7968DCEE813EC:C0396B8309F7EF9D]:0)
11:54:36         at org.junit.Assert.fail(Assert.java:86)
11:54:36         at org.junit.Assert.assertTrue(Assert.java:41)
11:54:36         at org.junit.Assert.assertTrue(Assert.java:52)
11:54:36         at org.elasticsearch.xpack.searchablesnapshots.cache.common.CacheFileTests.testCacheFileCreatedAsSparseFile(CacheFileTests.java:422)
```

**Comment by przemekwitek** [comment]
@Leroux: You seem to have modified this test today. Could you take a look?

**Comment by ywangd** [comment]
This test [failed again](https://gradle-enterprise.elastic.co/s/mm6twk4w5slek/tests/:x-pack:plugin:searchable-snapshots:test/org.elasticsearch.xpack.searchablesnapshots.cache.common.CacheFileTests/testCacheFileCreatedAsSparseFile?expanded-stacktrace=WyIwIl0&top-execution=1). It has a different error message but I guess it could still be related to this issue. Also since this issue is relatively new. I am re-openning it.

```
java.lang.AssertionError: Cache file size mismatches (block size: 8192, number of blocks: 97) |  
-- | --
  | Expected: <794624L> |  
  | but: was <798720L>
```

**Comment by tlrx** [comment]
Thanks for reopening - the failure is indeed different and only occurs on encrypted at rest builds.

I opened #81527 for this.

**Comment by dimitris-athanasiou** [comment]
It seems we got another related failure. This time it was:

```


java.lang.AssertionError: Non default block size only used in test executed with encryption at rest |  
-- | --
  | Expected: a string containing "/mnt/secret" |  
  | but: was "/dev/shm/elastic+elasticsearch+master+multijob+platform-support-arm/x-pack/plugin/searchable-snapshots/build/testrun/test/temp/org.elasticsearch.xpack.searchablesnapshots.cache.common.CacheFileTests_86F3C242690A78-001/tempDir-010/BBqS1___T_-9xbA_AAAAAA"


```

**Comment by tlrx** [comment]
Last failure should be fixed by #81748 🤞🏻
